src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -347,15 +378,56 @@ public async Task InvalidateCachedDataBlocks(Guid releaseId)
                 .CreateCacheKeyForDataBlock(releaseId, dataBlockId)
                 .OnSuccessVoid(_cacheService.DeleteItem);
         }
+
+        public async Task<Either<ActionResult, List<DataBlockViewModel>>> GetUnattachedDataBlocks(Guid releaseId)
+        {
+            return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanViewRelease)
+                .OnSuccess(async release =>
+                {
+                    var keyStatDataBlockIds = await _context.KeyStatisticsDataBlock
+                        .Where(ks => ks.ReleaseId == release.Id)
+                        .Select(ks => ks.DataBlockId)
+                        .ToListAsync();
+
+                    return await _context
+                        .ReleaseContentBlocks
+                        .Include(releaseContentBlock => releaseContentBlock.ContentBlock)
+                        .Where(releaseContentBlock => releaseContentBlock.ReleaseId == release.Id)
+                        .Select(releaseContentBlock => releaseContentBlock.ContentBlock)
+                        .OfType<DataBlock>()
+                        .ToAsyncEnumerable()
+                        .WhereAwait(async dataBlock =>
+                            await IsUnattachedDataBlock(dataBlock, keyStatDataBlockIds))
+                        .OrderBy(contentBlock => contentBlock.Name)
+                        .Select(contentBlock => _mapper.Map<DataBlockViewModel>(contentBlock))
+                        .ToListAsync();
+                });
+        }
+
+        public async Task<bool> IsUnattachedDataBlock(DataBlock dataBlock, List<Guid>? keyStatDataBlockIds = null)
+        {
+            if (keyStatDataBlockIds == null)
+            {
+                return dataBlock.ContentSectionId == null
+                       && await _context.KeyStatisticsDataBlock.AllAsync(ks =>
+                           ks.DataBlockId != dataBlock.Id);
+            }
+
+            return dataBlock.ContentSectionId == null
+                   && await keyStatDataBlockIds
+                       .ToAsyncEnumerable()
+                       .AllAsync(dataBlockId => dataBlockId != dataBlock.Id);
+        }
```

Comment:
Discussed this on Slack. I've used the solution Ben suggests above - this does mean that the query gets made for each datablock iterated over in the `WhereAwait` in `DataBlockService#GetUnattachedDataBlocks`. But decided the benefits in readability and maintainability trump the performance gain. This can always be fixed if it turns into an issue - and it would only affect Admin, so it's not a big risk.